Proposed_First_FLEDGE_OT_Details.md

```diff
@@ -0,0 +1,172 @@
+
+#### Trusted Signals
+
+The FOT#1 will include browser support for fetching trusted bidding and scoring signals as documented in the explainer. These signals are fetched prior to the corresponding bidding or scoring worklet being executed; the fetched signals are then passed into the corresponding worklet.
+
+  
+
+The FOT#1 follows a “Bring Your Own Server” (BYOS) model as the FLEDGE explainer talks about: [“As a temporary mechanism during the First Experiment timeframe, the buyer and seller can fetch these bidding signals from any server, including one they operate themselves (a "Bring Your Own Server" model).”](https://github.com/WICG/turtledove/blob/main/FLEDGE.md#31-fetching-real-time-data-from-a-trusted-server) Allowing a BYOS model initially helps [promote accurate measurement](#promoting-accurate-measurement-of-fledge-effectiveness.) by not conflating FLEDGE performance with complexities of operating a trusted server, and is due to availability of accepted server trust technologies.
+
+#### Interest Group Updating
+
+The FOT#1 will include a manual interest group update API, `navigator.updateAdInterestGroups()` that updates interest groups owned by the calling frame that haven’t been updated in the last 24 hours. This updating will be done using the interest group’s `dailyUpdateUrl` setting as the explainer discusses.
+
+  
+
+The FOT#1 will also include a mechanism to update interest groups that have just had the chance to participate in an auction initiated by `navigator.runAdAuction()`. The updating will happen shortly after the auction and will update interest groups that haven’t been updated in the last 24 hours. This updating will be done using the interest group’s `dailyUpdateUrl` setting as the explainer discusses.
+
+  
+
+The FOT#1 will not include a periodic background interest group update mechanism.
+
+#### Advertisement Rendering
+
+The FOT#1 will include an implementation of `navigator.runAdAuction()` that returns Promises that resolve to URLs that are opaque to the caller, as the explainer discusses.
+
+  
+
+The FOT#1 will include temporary support for rendering these opaque URLs in iframes. This [promotes accurate measurement](#promoting-accurate-measurement-of-fledge-effectiveness.) by not conflating FLEDGE performance with Fenced Frame rendering differences.
+
+  
+
+In the FOT#1, rendering these opaque URLs only in Fenced Frames will temporarily not be required, and they can instead be rendered in iframes as mentioned above. When [Fenced Frames are available for testing](#availability-of-other-technologies.), these opaque URLs will be renderable in Fenced Frames. At a later date, the ability to render opaque URLs in iframes will be removed.
```

Comment:
How will this interact with CSP?  If I have `<iframe src="urn:uuid:1234abc...">` will this be (a) evaluated against my CSP as a `urn:uuid:` URL, (b) evaluated as the underlying `https:` URL, or (c) treated as planned for fenced frames?

If (a) it seems like adtech vendors will need to use `navigator.deprecatedURNToURL()` to get an `https:` URL, unless they're sure pages permit `urn:uuid:` sources for iframes.

Reply:
@brusshamilton, author of https://crrev.com/948399, do you know how URNs in iframes interact with CSP?